### DISCUSSION TITLE
Promoted attribute template?

### DISCUSSION THREAD

**Discussion by jobidon** [TriliumNext/Trilium#3791] [General]
Is there a way to customize the layout of promoted attributes? Templates can be used for the title and the content, but is there a  type of template specifically for promoted attributes?

The use-case is I am defining a note with 10+ inherited promoted attributes and instead of having the software guess how to lay them out in the note header, I'd like to use a html fragment/template to specify how to render/display them. I understand how in some cases an attribute might not be included in the template, so the ability to lay them out automatically would still need to be retained. So perhaps instead of promoting the attributes, I could create a note template with JS to populate un-promoted attributes and use a note template to manage those attributes that are important, and let the rest (promoted attributes) be displayed in the header as the software determines it best?

Or is there a better way to do this?

**Comment by zadam** [comment]
There isn't any built-in way to do this. With JS scripting / [custom widgets](https://github.com/zadam/trilium/wiki/Custom-widget) you have absolute freedom, but it's of course much more involved.
